src/beacon/signatures/mod.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod public_key_impls;
+mod signature_impls;
+
+use bls_signatures::Error;
+use blstrs::{G1Affine, G1Projective, G2Affine, G2Projective};
+use group::{prime::PrimeCurveAffine, Curve};
+use rayon::prelude::*;
+
+// re-exports
+pub use bls_signatures::{PublicKey as PublicKeyOnG1, Signature as SignatureOnG2};
+
+const CSUITE_G1: &[u8] = b"BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_";
+const CSUITE_G2: &[u8] = b"BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_";
```

Comment:
Where do these come from?

Reply:
https://www.ietf.org/archive/id/draft-irtf-cfrg-bls-signature-05.html#name-basic

Reply:
Added to the comment